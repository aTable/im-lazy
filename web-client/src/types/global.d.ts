declare global {
    interface Window {
        mgr: Oidc.UserManager // TODO: doesnt work?
        MyNamespace: any
    }
}
