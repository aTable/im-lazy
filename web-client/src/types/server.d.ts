export interface IWeatherForecast {
    date: Date
    temperatureC: number
    temperatureF: number
    summary: string
}
