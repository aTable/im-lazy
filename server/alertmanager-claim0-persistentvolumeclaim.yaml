apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  creationTimestamp: 2020-10-02T00:00:00Z
  labels:
    io.kompose.service: alertmanager-claim0
  name: alertmanager-claim0
spec:
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 100Mi
status: {}
