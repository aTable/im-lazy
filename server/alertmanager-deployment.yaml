apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    kompose.cmd: kompose convert
    kompose.version: 1.21.0 (992df58d8)
  creationTimestamp: 2020-10-02T00:00:00Z
  labels:
    io.kompose.service: alertmanager
  name: alertmanager
spec:
  replicas: 1
  selector:
    matchLabels:
      io.kompose.service: alertmanager
  strategy:
    type: Recreate
  template:
    metadata:
      annotations:
        kompose.cmd: kompose convert
        kompose.version: 1.21.0 (992df58d8)
      creationTimestamp: 2020-10-02T00:00:00Z
      labels:
        io.kompose.service: alertmanager
    spec:
      containers:
      - args:
        - --config.file=/etc/alertmanager/config.yml
        - --storage.path=/alertmanager
        image: prom/alertmanager:latest
        imagePullPolicy: Always
        name: alertmanager
        ports:
        - containerPort: 9093
        resources: {}
        volumeMounts:
        - mountPath: /etc/alertmanager/
          name: alertmanager-claim0
      restartPolicy: Always
      serviceAccountName: ""
      volumes:
      - name: alertmanager-claim0
        persistentVolumeClaim:
          claimName: alertmanager-claim0
status: {}
