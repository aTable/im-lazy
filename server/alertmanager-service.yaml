apiVersion: v1
kind: Service
metadata:
  annotations:
    kompose.cmd: kompose convert
    kompose.version: 1.21.0 (992df58d8)
  creationTimestamp: 2020-10-02T00:00:00Z
  labels:
    io.kompose.service: alertmanager
  name: alertmanager
spec:
  ports:
  - name: "9086"
    port: 9086
    targetPort: 9093
  selector:
    io.kompose.service: alertmanager
status:
  loadBalancer: {}
