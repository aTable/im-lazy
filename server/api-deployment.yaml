apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  annotations:
    kompose.cmd: kompose convert
    kompose.version: 1.16.0 (0c01309)
  creationTimestamp: null
  labels:
    io.kompose.service: api
  name: api
spec:
  replicas: 1
  strategy:
    type: Recreate
  template:
    metadata:
      creationTimestamp: null
      labels:
        io.kompose.service: api
    spec:
      containers:
      - env:
        - name: ASPNETCORE_ENVIRONMENT
          value: Development
        image: api
        name: yournamespaceapi
        ports:
        - containerPort: 9080
        - containerPort: 9081
        resources: {}
        volumeMounts:
        - mountPath: /https
          name: api-claim0
      restartPolicy: Always
      volumes:
      - name: api-claim0
        persistentVolumeClaim:
          claimName: api-claim0
status: {}
