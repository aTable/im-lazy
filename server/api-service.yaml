apiVersion: v1
kind: Service
metadata:
  annotations:
    kompose.cmd: kompose convert
    kompose.version: 1.16.0 (0c01309)
  creationTimestamp: null
  labels:
    io.kompose.service: api
  name: api
spec:
  ports:
  - name: "9080"
    port: 9080
    targetPort: 9080
  - name: "9081"
    port: 9081
    targetPort: 9081
  selector:
    io.kompose.service: api
status:
  loadBalancer: {}
