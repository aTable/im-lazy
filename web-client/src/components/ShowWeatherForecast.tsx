import React from 'react'
import { IWeatherForecast } from '../types/server'

export interface IShowWeatherForecastProps {
    weather?: IWeatherForecast[]
}

const ShowWeatherForecast = (props: IShowWeatherForecastProps) => {
    return (
        <table className="table">
            <thead>
                <tr>
                    <th>Date</th>
                    <th>Summary</th>
                    <th>Temp (C)</th>
                    <th>Temp (F)</th>
                </tr>
            </thead>
            <tbody>
                {!props.weather || (props.weather.length <= 0 && <p>No weather data</p>)}
                {props.weather?.map(x => (
                    <tr key={x.date.toString()}>
                        <td>{x.date.toString()}</td>
                        <td>{x.summary}</td>
                        <td>{x.temperatureC}</td>
                        <td>{x.temperatureF}</td>
                    </tr>
                ))}
            </tbody>
        </table>
    )
}

export default ShowWeatherForecast
