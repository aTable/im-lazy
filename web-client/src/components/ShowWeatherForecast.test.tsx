import React from 'react'
import ReactDOM from 'react-dom'
import ShowWeatherForecast from './ShowWeatherForecast'
import renderer from 'react-test-renderer'
import { IWeatherForecast } from '../types/server'

it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<ShowWeatherForecast />, div)
    ReactDOM.unmountComponentAtNode(div)
})

it('renders loading', () => {
    const component = renderer.create(<ShowWeatherForecast weather={[]} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})

it('renders forecast', () => {
    const weather: IWeatherForecast[] = [
        {
            date: new Date(0),
            summary: 'dis b hot',
            temperatureC: 1000,
            temperatureF: 1000,
        },
    ]
    const component = renderer.create(<ShowWeatherForecast weather={weather} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
