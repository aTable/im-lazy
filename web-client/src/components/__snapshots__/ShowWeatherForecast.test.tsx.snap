// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders forecast 1`] = `
<table>
  <thead>
    <tr>
      <th>
        Date
      </th>
      <th>
        Summary
      </th>
      <th>
        Temp (C)
      </th>
      <th>
        Temp (F)
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Thu Jan 01 1970 10:00:00 GMT+1000 (Australian Eastern Standard Time)
      </td>
      <td>
        dis b hot
      </td>
      <td>
        1000
      </td>
      <td>
        1000
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`renders loading 1`] = `
<p>
  No weather data
</p>
`;
